packages/graphql/lib/schema-builder/graphql-schema.factory.ts

```diff
@@ -60,7 +60,6 @@ export class GraphQLSchemaFactory {
       options = scalarsOrOptions;
     }
 
-    TypeMetadataStorage.clear();
```

Comment:
Why this line is no longer required?

Reply:
Because we save the metadata in maps. 
What I think happened was that when we used array to save the metadata, and called updateStorage in lazy-metadata.storage multiple times, the metadata would get duplicated in the storage.

For performance optimization, I removed redundant updateStorage calls, but now calling GraphQLSchemaFactory.create called for TypeMetadataStorage.clear (it expected the the metadata would be collected again via updateStorage), and the metadata would not get re-collected (because I removed duplicate updateStorage calls).

So I removed the clear call. 

Why this works?
1) Since TypeMetadataStorage has changed to maps, there is no chance of getting duplicate metadata values. 
2) When we have no duplicates, there is no need to reset and re-collect the metadata.

I hope my explanation is clear.

Reply:
The only thing that bothers me, is that TypeMetadataStorage is a singleton that is generated on app bootstrap, and is not bounded by a NestJS instance